### PR TITLE
Adding store_failures_as

### LINKED ISSUE #4246: Support test results as a view dbt-core PR #8653
https://github.com/dbt-labs/dbt-core/pull/8653

At a minimum, update body to include a link to the page on docs.getdbt.com requiring updates and what part(s) of the page you would like to see updated.

Pages:
- update https://docs.getdbt.com/docs/build/tests#storing-test-failures
- create a new page similar to https://docs.getdbt.com/reference/resource-configs/store_failures
- maybe some other places too? 🤷

### PR DESCRIPTION
**Pull request by matthewshaver**
## What are you changing in this pull request and why?

Closes: https://github.com/dbt-labs/docs.getdbt.com/issues/4246

Adding a page for resource config `store_failures_as`

## Checklist

- [x] Add versioning components, as described in [Versioning Docs](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-entire-pages)
- [x] Add a note to the prerelease version [Migration Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/guides/migration/versions)

- [x] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [x] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding new pages (delete if not applicable):
- [x] Add page to `website/sidebars.js`
- [x] Provide a unique filename for the new page

### CODE REVIEW

File: website/docs/docs/build/tests.md [hunk 1]
```diff
@@ -241,7 +241,7 @@ where {{ column_name }} is null
 
 ## Storing test failures
 
-Normally, a test query will calculate failures as part of its execution. If you set the optional `--store-failures` flag or [`store_failures` config](/reference/resource-configs/store_failures), dbt will first save the results of a test query to a table in the database, and then query that table to calculate the number of failures.
+Normally, a test query will calculate failures as part of its execution. If you set the optional `--store-failures` flag or the [`store_failures`](/reference/resource-configs/store_failures), or [`store_failures_as`](/reference/resource-configs/store_failures_as) configs, dbt will first save the results of a test query to a table in the database, and then query that table to calculate the number of failures.
```
Comment (matthewshaver) [inline website/docs/docs/build/tests.md]:
```suggestion
Normally, a test query will calculate failures as part of its execution. If you set the optional `--store-failures` flag,  the [`store_failures`](/reference/resource-configs/store_failures), or the [`store_failures_as`](/reference/resource-configs/store_failures_as) configs, dbt will first save the results of a test query to a table in the database, and then query that table to calculate the number of failures.
```

File: website/docs/reference/resource-configs/store_failures_as.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
+---
+resource_types: [tests]
+id: "store_failures_as"
+---
+
+For the `test` resource type, `store_failures_as` is an optional config that specifies how test failures should be stored in the database. If [`store_failures`](/reference/resource-configs/store_failures) is configured, `store_failures_as` takes precedence.
```
Comment (runleonarun) [inline website/docs/reference/resource-configs/store_failures_as.md]:
```suggestion
For the `test` resource type, `store_failures_as` is an optional config that specifies how test failures should be stored in the database. If [`store_failures`](/reference/resource-configs/store_failures) is also configured, `store_failures_as` takes precedence. 
```

File: website/docs/reference/resource-configs/store_failures_as.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
+---
+resource_types: [tests]
+id: "store_failures_as"
+---
+
+For the `test` resource type, `store_failures_as` is an optional config that specifies how test failures should be stored in the database. If [`store_failures`](/reference/resource-configs/store_failures) is configured, `store_failures_as` takes precedence. 
+
+The three supported values are:
```
Comment (runleonarun) [inline website/docs/reference/resource-configs/store_failures_as.md:8]: Where are they stored by default?

**Review by runleonarun — APPROVED** [review]
Looks good! Only one question about default which I think is table, but not sure?

**Pull request merged**